### ISSUE TITLE
Hikari dropping all connections

### ISSUE THREAD

**Issue by wesoos** [brettwooldridge/HikariCP#1542]
Hello,  I've had Hikari running for years, mostly without issue.  However since yesterday we are experiencing a major issue where Hikari, after running for a few hours, drops all connections, and does not recreate them.  I'm trying to go through the logs to find a specific error, but my logs are littered right now from my application trying to make database requests but can't.

I restart my application and it runs fine for a while, then all connections just closes.

I do have leak detection enabled with 2minutes, and do see some of those, but they are somewhat expected because I run some big reports sometimes.  

Please advise!

**Comment by wesoos** [comment]
This started happening out of the blue, btw, I did not change anything in my code...

**Comment by wesoos** [comment]
Database is MySQL.  App is spring boot based...

**Comment by wesoos** [comment]
Going through the logs.  After it closed all the connections, it repeatedly logs this message, with varying "waiting" connections.

HikariPool-1 - Timeout failure stats (total=0, active=0, idle=0, waiting=11)
--

Then after a couple hours, I get this message:

HikariPool-1 - Failed to create/setup connection: The last packet successfully received from the server was 7,884,451 milliseconds ago.  The last packet sent successfully to the server was 7,884,451 milliseconds ago. is longer than the server configured value of 'wait_timeout'. You should consider either expiring and/or testing connection validity before use in your application, increasing the server configured values for client timeouts, or using the Connector/J connection property 'autoReconnect=true' to avoid this problem.
--

Then all of a sudden it starts adding connections again.

Please help

**Comment by billoneil** [comment]
See if there are any mismatches with timeouts. https://github.com/brettwooldridge/HikariCP/issues/1228#issuecomment-590860636

**Comment by wesoos** [comment]
Yep they are good.  I do have a proxy between the pool and database.  The proxy timeout is 5 minutes, and my pool maxLifetime is 4 minutes.  So that shouldn't be an issue...

**Comment by brettwooldridge** [comment]
@wesoos What version of HikariCP are you running?  What version of the MySQL or MariaDB driver are you using?

If you do see leak messages, do you also see matching "unleak" messages.  These should have the form:
```
Previously reported leaked connection {} on thread {} was returned to the pool (unleaked)
```
